api/envoy/api/v2/core/base.proto

```diff
@@ -112,7 +141,26 @@ message Node {
   // This is motivated by informing a management server during canary which
   // version of Envoy is being tested in a heterogeneous fleet. This will be set
   // by Envoy in management server RPCs.
-  string build_version = 5;
+  // Deprecated in favor of the user_agent_name and user_agent_version values.
+  string build_version = 5 [deprecated = true];
+
+  // Free-form string that identifies the entity requesting config.
+  // E.g. "envoy" or "grpc"
+  string user_agent_name = 6;
+
+  // Free-form string that identifies the version of the entity requesting config.
+  // E.g. "1.12.2" or "abcd1234", or "SpecialEnvoyBuild"
+  string user_agent_version = 7;
+
+  // List of extensions and their versions supported by the node.
+  repeated Extension extensions = 8;
```

Comment:
Can we make this optional and send only when requested for via some field in `config_source` similar to `https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/core/config_source.proto#L72`  like `send_extensions_in_node` or some thing similar?

Reply:
IIUC, this means that you would get the Node on every request but that the node would not contain the extensions information on every request. What's the rationale for that?

Reply:
Yeah, I think `set_node_on_first_message_only` covers this use case already, unless there is something missed.

Reply:
> What's the rationale for that?

It is mainly for folks who are not using that field `set_node_on_first_message_only` as that field was added recently and is an optional one. Their management server implementations have not set that field and hence would get `Node` on every message and this could impact their performance.  If people find this causing perf issue, they will have to change their management server implementations.

Reply:
@ramaraochavali I think the change to the xDS protocol you are suggesting is somewhat outside of the scope of this change. I think we will deal with this later if the performance issue comes up.